src/clojure_lsp/main.clj

```diff
@@ -430,6 +431,24 @@
         (end
           (apply #'handlers/extension method args)))))
 
+(defn process-alive?
+  [pid]
+  (let [{:keys [exit]} (shell/sh "kill" "-0" (str pid))]
+    (= exit 0)))
+
+(defn start-parent-process-liveness-probe!
+  [ppid server]
+  (future (run!
```

Comment:
I think you can use `go` macro since it uses clojure.async/go which is async too but is lighter than another thread, WDYT?

Reply:
We already use that macro for almost all methods of this namespace

Reply:
Yeah, I thought about this but wasn't sure which is better for this, given that async `go` runs on the same thread but parks it, whereas the future runs in a new thread. I'm not sure if the use of `Thread/sleep` makes a difference in which one we should use. WDYT?

Reply:
I guess what I mean is will it sleep the thread and block other operations if we used async `go`? Worth a try to see if we don't know, I suppose.

Reply:
Oh I see we can use `<!` with `timeout`. I'm still unsure what the implications of each is (parking vs new thread for this usage), but you may be more familiar with this.

Reply:
Hum, true, probably it will block the thread.. I think it's okay to go with the future

Reply:
Oh, yes, probably a timeout may solve that, something like:
```clojure
(defun check-process! []
  (async/go
    (async/<! (async/timeout 1000))
    (if (alive?)
      (check-process!)
      (exit))))
```